test/unit/receivers/SocketModeReceiver.spec.ts

```diff
@@ -650,4 +655,72 @@ describe('SocketModeReceiver', () => {
       assert(clientStub.disconnect.called);
     });
   });
+
+  describe('event', () => {
+    it('acknowledges processed events', async () => {
+      const processStub = sinon.stub<[ReceiverEvent]>().resolves();
+      processStub.callsFake(async (event: ReceiverEvent) => {
+        await event.ack();
+      });
+      const app = sinon.createStubInstance(App, { processEvent: processStub }) as unknown as App;
+      const SocketModeReceiver = await importSocketModeReceiver(overrides);
+      const receiver = new SocketModeReceiver({ appToken: 'xapp-example' });
+      receiver.init(app);
+
+      // Stub ack with an awaited promise for tests
+      let resolve: () => void;
+      const called = new Promise<void>((r) => {
+        resolve = r;
+      });
+      const ack = sinon.stub().callsFake(async () => {
+        resolve();
+      });
+
+      receiver.client.emit('slack_event', { ack });
+      await called;
+      assert.isTrue(ack.called);
+    });
+
+    it('acknowledges events that throw AuthorizationError', async () => {
+      const processStub = sinon.stub<[ReceiverEvent]>().throws(new AuthorizationError('brokentoken', new Error()));
+      const app = sinon.createStubInstance(App, { processEvent: processStub }) as unknown as App;
+      const SocketModeReceiver = await importSocketModeReceiver(overrides);
+      const receiver = new SocketModeReceiver({ appToken: 'xapp-example' });
+      receiver.init(app);
+
+      // Stub ack with an awaited promise for tests
+      let resolve: () => void;
+      const called = new Promise<void>((r) => {
+        resolve = r;
+      });
+      const ack = sinon.stub().callsFake(async () => {
+        resolve();
+      });
+
+      receiver.client.emit('slack_event', { ack });
+      await called;
+      assert.isTrue(ack.called);
+    });
+
+    it('does not acknowledge events that throw unknown errors', async () => {
+      const processStub = sinon.stub<[ReceiverEvent]>().throws(new Error('internal error'));
+      const defaultProcessEventErrorHandlerSpy = sinon.spy(defaultProcessEventErrorHandler);
+      const app = sinon.createStubInstance(App, { processEvent: processStub }) as unknown as App;
+      const SocketModeReceiver = await importSocketModeReceiver(overrides);
+      const receiver = new SocketModeReceiver({
+        appToken: 'xapp-example',
+        processEventErrorHandler: defaultProcessEventErrorHandlerSpy,
+      });
+      receiver.init(app);
+
+      const slackRequestArgs = { body: {}, ack: sinon.fake() };
+
+      receiver.client.emit('slack_event', slackRequestArgs);
+      while (!defaultProcessEventErrorHandlerSpy.called) {
+        await delay(50);
+      }
```

Comment:
🧠 This is a super good case to cover too! We can now be more confident in retries on erroring cases I believe.